lib/database/migrations/20240718120539-add-is-physics-column-to-detectors.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+'use strict';
+
+const { Sequelize } = require('sequelize');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+    up: async (queryInterface) => queryInterface.sequelize.transaction(async (transaction) => {
+        await queryInterface.addConstraint('run_detectors', {
+            type: 'FOREIGN KEY',
+            fields: ['detector_id'],
+            name: 'detector_id_fk_run_detectors',
+            references: {
+                table: 'detectors',
+                field: 'id',
+            },
+        }, { transaction });
+
+        await queryInterface.addConstraint('run_detectors', {
+            type: 'FOREIGN KEY',
+            fields: ['run_number'],
+            name: 'run_number_fk_run_detectors',
+            references: {
+                table: 'runs',
+                field: 'run_number',
+            },
+        }, { transaction });
+
+        await queryInterface.addColumn('detectors', 'is_physical', { type: Sequelize.BOOLEAN }, { transaction });
+        await queryInterface.sequelize.query(`
+        UPDATE detectors
+            SET is_physical = true
+    `, { transaction });
+
+        await queryInterface.sequelize.query(`
+            INSERT INTO detectors (name, is_physical)
+                SELECT name, False FROM dpl_detectors
+                    WHERE name NOT IN (SELECT name FROM detectors)
```

Comment:
This will be for when you actually move the data I suppose, not in this migration yet

Reply:
applied